hack/check/misspell.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+TF_REPO_PATH="$(git rev-parse --show-toplevel)"
+
+MISSPELL_LOC="${TF_REPO_PATH}/hack/check/tools/bin"
+
+# Install tools we need if it is not present
+if [[ ! -f "${MISSPELL_LOC}/misspell" ]]; then
+  curl -L https://git.io/misspell | bash
```

Comment:
For some reason this is failing for me on an M1 macOS system. But since we don't currently support that platform officially, I think this is fine. Looks likely an issue with the misspell project not supporting it yet.